kpops/cli/pipeline_config.py

```diff
@@ -1,11 +1,18 @@
+from __future__ import annotations
+
 from pathlib import Path
+from typing import TYPE_CHECKING, Any
 
 from pydantic import BaseConfig, BaseSettings, Field
-from pydantic.env_settings import SettingsSourceCallable
 
 from kpops.component_handlers.helm_wrapper.model import HelmConfig, HelmDiffConfig
 from kpops.utils.yaml_loading import load_yaml_file
 
+if TYPE_CHECKING:
```

Comment:
why is this necessary?

Reply:
See: [TCH002](https://beta.ruff.rs/docs/rules/typing-only-third-party-import/), [TCH003](https://beta.ruff.rs/docs/rules/typing-only-standard-library-import/)

Reply:
are you using Ruff locally already? 😄

Reply:
I have been for some time now :grin: 

I like it a lot so far

Reply:
nice, looking forward to adopting it for the project